spec/defines/server_module_spec.rb

```diff
@@ -16,108 +16,113 @@
-          mandatory_params.merge({:incoming_chmod => false,
-                                  :outgoing_chmod => false,
-          })
+          mandatory_params.merge(incoming_chmod: false,
+                                 outgoing_chmod: false)
         end
-        it { is_expected.not_to contain_file(fragment_name).with_content(/^incoming chmod.*$/) }
-        it { is_expected.not_to contain_file(fragment_name).with_content(/^outgoing chmod.*$/) }
+
+        it { is_expected.not_to contain_file(fragment_name).with_content(%r{^incoming chmod.*$}) }
+        it { is_expected.not_to contain_file(fragment_name).with_content(%r{^outgoing chmod.*$}) }
       end
 
       {
-        :comment            => 'super module !',
-        :read_only          => 'no',
-        :write_only         => 'yes',
-        :use_chroot         => 'no',
-        :list               => 'no',
-        :uid                => '4682',
-        :gid                => '4682',
-        :numeric_ids        => 'no',
-        :incoming_chmod     => '0777',
-        :outgoing_chmod     => '0777',
-        :max_connections    => '10',
-        :timeout            => '10',
-        :secrets_file       => '/path/to/secrets',
-        :hosts_allow        => ['localhost', '169.254.42.51'],
-        :hosts_deny         => ['some-host.example.com', '10.0.0.128'],
-        :transfer_logging   => 'true',
-        :log_format         => '%t %a %m %f %b',
-        :refuse_options     => ['c', 'delete'],
-        :include            => ['a', 'b'],
-        :exclude            => ['c', 'd'],
-        :include_from       => '/path/to/file',
-        :exclude_from       => '/path/to/otherfile',
-        :dont_compress      => ['a', 'b'],
-        :ignore_nonreadable => 'yes'
-      }.each do |k,v|
+        comment:            'super module !',
+        read_only:          'no',
+        write_only:         'yes',
+        use_chroot:         'no',
+        list:               'no',
+        uid:                '4682',
+        gid:                '4682',
+        numeric_ids:        'no',
+        incoming_chmod:     '0777',
+        outgoing_chmod:     '0777',
+        max_connections:    '10',
+        timeout:            '10',
+        secrets_file:       '/path/to/secrets',
+        hosts_allow:        ['localhost', '169.254.42.51'],
+        hosts_deny:         ['some-host.example.com', '10.0.0.128'],
+        transfer_logging:   'true',
+        log_format:         '%t %a %m %f %b',
+        refuse_options:     ['c', 'delete'],
+        include:            ['a', 'b'],
+        exclude:            ['c', 'd'],
+        include_from:       '/path/to/file',
+        exclude_from:       '/path/to/otherfile',
+        dont_compress:      ['a', 'b'],
+        ignore_nonreadable: 'yes',
+      }.each_pair do |k, v|
         describe "when overriding #{k}" do
           let :params do
-            mandatory_params.merge({ k => v })
+            mandatory_params.merge(Hash[k, v])
```

Comment:
I think this change is not correct. The style guide prefers the first https://rubystyle.guide/#avoid-hash-constructor

Reply:
A problem here is RuboCop doesn't allow usage of { k => v } because it's old style dict. I initially tried { k: v } but this doesn't work as intended (seems passing variables doesn't work with new style of dict) and ended up with that ugly hack. May I ask your suggestion here ?

Reply:
Curious... which cop is complaining here? The new-style hash syntax only applies to symbol keys; for hashes with unknown key types or non-symbol key types, the hash-rocket syntax is still correct. I am surprised to hear that RuboCop objects to `mandatory_params.merge({ k => v })`

Reply:
This is what I get when I revert this change.

`pdk (CONVENTION): rubocop: Style/BracesAroundHashParameters: Redundant curly braces around a hash parameter. (spec/defines/server_module_spec.rb:106:36)`

Reply:
Oh, wait. I misunderstood what Rubocop complains. It's refusing not complete format but only {}. I'll fix this ASAP.